src/core.ts

```diff
@@ -124,6 +132,30 @@ export function instrument(eventState: IInstrumentationEventState) {
   }
 }
 
+export function onWorkerMessage(evt: MessageEvent) {
+  if (state !== State.Unloaded) {
+    let message = evt.data;
+    switch (message.type) {
+      case WorkerMessageType.Upload:
+        let uploadMsg = message as IUploadMessage;
+        let onSuccess = (status: number) => { mapProperties(droppedPayloads, uploadDroppedPayloadsMappingFunction, true); };
+        let onFailure = (status: number) => { onFirstSendDeliveryFailure(status, uploadMsg.rawData, uploadMsg.compressedData); };
+        upload(uploadMsg.compressedData, onSuccess, onFailure);
+        pendingEvents.splice(0, uploadMsg.eventCount);
```

Comment:
>splice [](start = 22, length = 6)

Suggest is to have something here that lets you see if it matches what is in the messages posted back from the worker and check the match.

Reply:
I could store pendingEvents as an object Map<eventId, eventObject>, and then parse batch returned by the worker to inspect all event IDs that it contains and then clear them from the pendingEvents storage 1 by 1.

This would be less brittle, but obviously more complicated.

---
In reply to: [137240630](https://github.com/Microsoft/clarity-js/pull/28#discussion_r137240630) [](ancestors = 137240630)